collectors/StatsCollector.py

```diff
@@ -0,0 +1,67 @@
+from BaseCollector import BaseCollector
+import logging
+
+logger = logging.getLogger('vrops-exporter')
+
+
+class StatsCollector(BaseCollector):
+
+    def __init__(self):
+        super().__init__()
+
+    def get_uuids(self):
+        raise NotImplementedError("Please Implement this method")
+
+    def set_labels(self, resource_id: str, project_ids: list):
+        raise NotImplementedError("Please Implement this method")
+
+    def collect(self):
+        logger.info(f'{self.name} starts with collecting the metrics')
+
+        token = self.get_target_tokens()
+        token = token.setdefault(self.target, '')
+        if not token:
+            logger.warning(f'skipping {self.target} in {self.name}, no token')
+            return
+
+        uuids = self.get_uuids()
+        metrics = self.generate_metrics(label_names=self.label_names)
+        project_ids = self.get_project_ids_by_target() if self.project_ids else []
+        values, api_responding = self.vrops.get_latest_stats_multiple(self.target,
+                                                                      token,
+                                                                      uuids,
+                                                                      [m for m in metrics],
+                                                                      self.name)
+        yield self.create_api_response_metric(self.name, api_responding)
+
+        if not values:
+            logger.warning(f'No values in the response for {self.name}. API code: {api_responding}')
+            return
+
+        no_match_in_config = set()
+        values_received = set()
+
+        for resource in values:
+            resource_id = resource.get('resourceId')
+            labels = self.set_labels(resource_id, project_ids)
+
+            for value_entry in resource.get('stat-list', {}).get('stat', []):
+                statkey = value_entry.get('statKey', {}).get('key')
+                values_received.add(statkey)
+
+                metric_data = value_entry.get('data', [None])[0]
+                if statkey in metrics and metric_data is not None:
+                    metrics[statkey]['gauge'].add_metric(labels=labels, value=metric_data)
+                else:
+                    no_match_in_config.add(statkey)
```

Comment:
What do you think of adding another metric, which could give us this information? It could only be created, if we are facing a "no match" and then we can at least easily visualise it in a dashboard. I still like the format here, but it will be buried in the logs. So either we have a kibana dashboard to visualise it as it is, or we will actually never see it.

Reply:
Have you seen the output? I case of virtual disks and datastores there are many metrics which doesn't match. I also thought about this solution you suggest. Let me fix this in a follow up PR.

Reply:
For an initial check it is good, then we still would need to go through every region to check on it. at least, initially.

Reply:
I implemented a workaround to get the metrics. I'm not happy with it yet, but at least we're getting it for now.